search_that_hash/__main__.py

```diff
@@ -72,20 +78,149 @@ def main(**kwargs):
     logging.debug("Updated logging level")
     logging.info("Called config updater")
 
+    #### UPDATING CONFIG
+
     config = config_object.cli_config(kwargs)
 
+    #### BANNER
+
     if not kwargs["greppable"] and not kwargs["accessible"] and not kwargs["no_banner"]:
         logging.info("Printing banner")
         printing.Prettifier.banner()
-    
-    searcher = cracking.Searcher(config)
-    results = cracking.Searcher.main(searcher)
-    
-    if kwargs["greppable"]:
-        logging.info("Printing greppable results")
-        printing.Prettifier.greppable_print(results)
+
+    #### CALLING CRACKING
+
+    cracking_handler(config)
 
     exit(0)
 
+
+def cracking_handler(config):
+
+    if config["api"]:
+        coloredlogs.install(level=logging.CRITICAL)
+
+    ### SETTING VARIBLES
+
+    hash_processes = []
+    results = []
+    searcher = cracking.Searcher(config)
+
+    #### STH CRACKING
+
+    sth_results, config = sth.crack(config)
+
+    if not sth_results == False:
+        for result in sth_results.values():
+            if not config["greppable"]:
+                if not config["api"]:
+                    printing.Prettifier.sth_print(
+                        result["Hash"],
+                        result["Plaintext"],
+                        result["Type"],
+                        result["Verified"],
+                    )
+                else:
+                    results.append({result["Hash"]: result["Plaintext"]})
+
+    #### MAIN CRACKING
+
+    for chash, types in config["hashes"].items():
+        if types == []:
+            if not config["greppable"]:
+                if config["api"]:
+                    results.append({chash: "No types found for this hash."})
+                    continue
+                printing.Prettifier.error_print("No types found for this hash.", chash)
+            continue
+
+        hash_processes.append(cracking.Searcher.main(searcher, chash, types))
+
+        #### OUTPUTTING
+
+        chash = list(hash_processes[-1].keys())[0]
+        result = hash_processes[-1][chash]
+
+        if result == None and not config["greppable"]:
+            if config["api"]:
+                results.append({chash: "Could not crack hash"})
+                continue
+            printing.Prettifier.error_print("Could not crack hash.", chash)
+            continue
+
+        if not config["greppable"]:
+            if config["api"]:
+                results.append({chash: result})
+                continue
+            printing.Prettifier.one_print(chash, result)
+
+    if config["greppable"]:
```

Comment:
You could have an entire class for greppable vs non instances so you don't need all these if statements everywhere?